Datagrid/ProxyQuery.php

```diff
@@ -113,11 +113,20 @@ public function execute(array $params = array(), $hydrationMode = null)
             ->getMetadataFor(current($queryBuilder->getRootEntities()))
             ->getIdentifierFieldNames();
 
+        $existsOrder = [];
+        /** @var Query\Expr\OrderBy $order */
+        foreach ($queryBuilder->getDQLPart('orderBy') as $order){
+            $existsOrder = array_merge($existsOrder, $order->getParts());
+        }
+
         foreach ($identifierFields as $identifierField) {
-            $queryBuilder->addOrderBy(
-                $rootAlias.'.'.$identifierField,
-                $this->getSortOrder() // reusing the sort order is the most natural way to go
-            );
+            $order = $rootAlias.'.'.$identifierField;
+            if(!in_array(($order.' '.$this->getSortOrder()), $existsOrder)){
```

Comment:
Not sure how it can actually happen, but IMO we are trying to prevent the field to be added twice, regardless of the field direction. `ORDER BY field1 `ASC, field2 DESC, field1DESC` makes no sense… plus, it would probably be more readable / easier to understand.